gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/AvroKeyRecursiveCombineFileInputFormat.java

```diff
@@ -69,15 +71,14 @@
   private static final String COMPACTION_JOB_MAPRED_MIN_SPLIT_SIZE = COMPACTION_JOB_PREFIX + "mapred.min.split.size";
   private static final long DEFAULT_COMPACTION_JOB_MAPRED_MIN_SPLIT_SIZE = 268435456;
 
-  private static final long GET_SPLIT_NUM_FILES_TRHESHOLD = 5000;
   private static final int SPLIT_MAX_NUM_LOCATIONS = 10;
-  private static final String AVRO = "avro";
 
   @Override
   public List<InputSplit> getSplits(JobContext cx) throws IOException {
-    setSplitSize(cx);
-    List<InputSplit> splits = getSplits(cx, Arrays.asList(getInputPaths(cx)));
-    return cleanSplits(splits);
+    Job modifiedJob = Job.getInstance(cx.getConfiguration());
+    setSplitSize(modifiedJob);
+    FileInputFormat.setInputDirRecursive(modifiedJob, true);
+    return cleanSplits(super.getSplits(modifiedJob));
```

Comment:
They aren't. Hidden files are automatically filtered, the assumption here is that all non-hidden files are avro files (normally a reasonable assumption for MR jobs).